taskcluster/ac_taskgraph/worker_types.py

```diff
@@ -6,16 +6,19 @@
 
 from six import text_type
 
-from voluptuous import Required, Optional
+from voluptuous import Any, Required, Optional
 
-from taskgraph.util.schema import taskref_or_string
+from taskgraph.util.schema import taskref_or_string, WHITELISTED_SCHEMA_IDENTIFIERS
 from taskgraph.transforms.task import payload_builder
 
+WHITELISTED_SCHEMA_IDENTIFIERS.extend([
+    lambda path: "[u'artifact-map']" in path,   # handed directly in beetmover
+])
```

Comment:
Is this necessary? If it is, can you file a bug about making this extension point more explicit?

Reply:
I believe it is. `artifact-map` is passed down verbatim and contains `taskId` (just like `upstream-artifacts` do). Without it, I get this error:

```
RuntimeError: YAML schemas should use dashed lower-case identifiers, not u'taskId' @ schema[u'artifact-map'][0][u'taskId']
```

What I don't understand is mozilla-central does the same, but doesn't have to declare it: 
* https://searchfox.org/mozilla-central/rev/b6f088f2f68d2a8ae0b49d6c8fbd7cbd3a65fa5b/taskcluster/taskgraph/util/schema.py#129
* https://searchfox.org/mozilla-central/rev/b6f088f2f68d2a8ae0b49d6c8fbd7cbd3a65fa5b/taskcluster/taskgraph/transforms/task.py#989
* https://searchfox.org/mozilla-central/rev/b6f088f2f68d2a8ae0b49d6c8fbd7cbd3a65fa5b/taskcluster/taskgraph/util/scriptworker.py#497


I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1586679, to keep track of it.

Reply:
This only looks at the schema, and not the contents of the payloads. Since m-c doesn't declare a schema for artifact maps (since it loads them from external files), it doesn't need to make an exception for them.

Reply:
Ah, that makes sense. Thanks for the explanation!